### PR TITLE
Add missing key destruction calls in ssl_write_client_key_exchange

### PR DESCRIPTION
**Pull request by AndrzejKurek**
Issue found in https://github.com/ARMmbed/mbedtls/pull/5525.
Backport: https://github.com/ARMmbed/mbedtls/pull/5577

### CODE REVIEW

File: library/ssl_cli.c [hunk 1]
```diff
@@ -3336,7 +3337,11 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
                                         own_pubkey, own_pubkey_max_len,
                                         &own_pubkey_len );
         if( status != PSA_SUCCESS )
+        {
+            if( psa_destroy_key( handshake->ecdh_psa_privkey ) == PSA_SUCCESS )
+                handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT;
```
Comment (superna9999) [inline library/ssl_cli.c:3342]: Not sure if the check hewre is really needed, I'd set `handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT` whatever thew result of `psa_destroy_key`

File: library/ssl_cli.c [hunk 2]
```diff
@@ -3351,12 +3356,12 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
...
-            return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );
 
-        status = psa_destroy_key( handshake->ecdh_psa_privkey );
-        if( status != PSA_SUCCESS )
+        destruction_status = psa_destroy_key( handshake->ecdh_psa_privkey );
+
+        if( status != PSA_SUCCESS || destruction_status != PSA_SUCCESS )
             return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );
+
         handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT;
```
Comment (superna9999) [inline library/ssl_cli.c:3365]: Perhaps `handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT` should be moved right after `psa_destroy_key`.

File: library/ssl_cli.c (hunk 1 above)
Comment (AndrzejKurek) [reply library/ssl_cli.c:3342]: I assumed that it should be set only in case of a succesfull destruction, as was the behaviour before.

Comment (superna9999) [reply library/ssl_cli.c:3342]: I do not know the current policy about that, setting the key id to `MBEDTLS_SVC_KEY_ID_INIT` after a key destroy isn't used a lot in the library

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Same remarks as Neil really.

**Review by gilles-peskine-arm — APPROVED** [review]

**Review by paul-elliott-arm — APPROVED** [review]
LGTM

**Comment by yanesca** [comment]
Is it possible to add a non-regression test for this?

**Review by superna9999 — APPROVED** [review]

**Comment by gilles-peskine-arm** [comment]
Upon further analysis we realized that this isn't a bug, just a robustness improvement, so https://github.com/Mbed-TLS/mbedtls/pull/5905 removes the changelog entry.

**Pull request merged**